ICSharpCode.ILSpyX/TreeView/PlatformAbstractions/IPlatformDragDrop.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace ICSharpCode.ILSpyX.TreeView.PlatformAbstractions
+{
+	public interface IPlatformDragDrop
+	{
+		XPlatDragDropEffects DoDragDrop(object dragSource, object data, XPlatDragDropEffects allowedEffects);
```

Comment:
Note: this and other operations should be async, as they are async in nature.

On desktop (and Browser with multithreading and hacks enabled), we can solve it by using DispatcherFrame to sync-wait on async operations. But since you are refactoring this part, consider making it async from the start as well.

Reply:
Note, DispatcherFrame is usable with both XPF and pure Avalonia. That's what I was using so far in new port too.

Reply:
Except mobile, Avalonia doesn't have any way to use DispatcherFrame on mobile yet. But I assume ILSpy won't benefit much from mobile support either way. Browser might be useful though.

Reply:
Ah, interesting - I was only reshuffling existing code, and there DoDragDrop was used in exactly that fashion.